src/main/java/com/jvm_bloggers/core/data_fetching/blogs/PreventConcurrentExecutionSafeguard.java

```diff
@@ -0,0 +1,32 @@
+package com.jvm_bloggers.core.data_fetching.blogs;
+
+import com.google.common.base.Supplier;
+
+import lombok.extern.slf4j.Slf4j;
+
+import java.util.concurrent.atomic.AtomicReference;
+
+@Slf4j
+public class PreventConcurrentExecutionSafeguard<T> {
```

Comment:
We don't. I did it by purpose to make the class more generic. I think we can face more problems like this in the future and then we can use the class OOTB.